src/recipe/tests/test_recipe.py

```diff
@@ -0,0 +1,112 @@
+from django.test import TestCase
+from django.urls import reverse
+from rest_framework import status
+from rest_framework.test import APIClient
+
+from core import models
+from core.models import Recipe, Ingredient
+from recipe.serializers import RecipeSerializer, IngredientSerializer
+
+RECIPE_URL = reverse('recipe:recipe-list')
+INGREDIENT_URL = reverse('recipe:ingredient-list')
+
+
+def ingredient_detail_url(ingredient_id):
+    """Return ingredient URL"""
+    return reverse('recipe:ingredient-detail', args=[ingredient_id])
+
+
+def add_ingredient_url(recipe_id):
+    """Return recipe add ingredient URL"""
+    return reverse('recipe:recipe-add-ingredient', args=[recipe_id])
+
+
+def sample_recipe(name='Recipe Read',
+                  description='Recipe Read Description'):
+    """Creates and persists a sample Recipe"""
+    recipe = models.Recipe.objects.create(
+        name=name,
+        description=description
+    )
+    recipe.save()
+    return recipe
+
+
+class IngredientViewTests(TestCase):
+    """Ingredient REST API Tests"""
+
+    def setUp(self):
+        self.client = APIClient()
```

Comment:
Aren't you missing `super().setUp()`?

Reply:
Good question, Django doesn’t uses it in the [documentation](https://docs.djangoproject.com/en/2.0/topics/testing/advanced/)

If we follow the parent clases the only declaration is in unittest.TestCase
```
def setUp(self):
    "Hook method for setting up the test fixture before exercising it."
    pass
```
Although the setUp method in the test is overriding its parent I don’t see it will be of any use other than following convention.

Let me know your thought.